pkg/api/v1alpha1/cluster_webhook_test.go

```diff
@@ -1960,6 +1802,57 @@ func TestClusterValidateUpdateSkipUpgradeImmutability(t *testing.T) {
 	}
 }
 
+func TestClusterValidateUpdateVersionSkew(t *testing.T) {
+	features.ClearCache()
+	cOld := createCluster()
+	cOld.Spec.ManagementCluster.Name = "mgmt2"
+	cOld.Spec.KubernetesVersion = "1.22"
+
+	cNew := cOld.DeepCopy()
+	cNew.Spec.KubernetesVersion = "1.24"
+	g := NewWithT(t)
+	err := cNew.ValidateUpdate(cOld)
+	g.Expect(err).To(MatchError(ContainSubstring("version difference between upgrade version (1.24) and server version (1.22) do not meet the supported version increment of +1")))
+}
+
+func TestClusterValidateUpdateVersionSkewDecrement(t *testing.T) {
+	features.ClearCache()
+	cOld := createCluster()
+	cOld.Spec.ManagementCluster.Name = "mgmt2"
+	cOld.Spec.KubernetesVersion = "1.24"
+
+	cNew := cOld.DeepCopy()
+	cNew.Spec.KubernetesVersion = "1.23"
+	g := NewWithT(t)
+	err := cNew.ValidateUpdate(cOld)
+	g.Expect(err).To(MatchError(ContainSubstring("version difference between upgrade version (1.23) and server version (1.24) do not meet the supported version increment of +1")))
+}
+
+func TestClusterValidateUpdateVersionInvalidNew(t *testing.T) {
+	features.ClearCache()
+	cOld := createCluster()
+	cOld.Spec.ManagementCluster.Name = "mgmt2"
+	cOld.Spec.KubernetesVersion = "1.24"
+
+	cNew := cOld.DeepCopy()
+	cNew.Spec.KubernetesVersion = "test"
+	g := NewWithT(t)
+	err := cNew.ValidateUpdate(cOld)
+	g.Expect(err).To(MatchError(ContainSubstring("parsing comparison version: could not parse \"test\" as version")))
+}
+
+func TestClusterValidateUpdateVersionInvalidOld(t *testing.T) {
+	features.ClearCache()
+	cOld := createCluster()
+	cOld.Spec.ManagementCluster.Name = "mgmt2"
+	cOld.Spec.KubernetesVersion = "test"
+
+	cNew := cOld.DeepCopy()
+	g := NewWithT(t)
+	err := cNew.ValidateUpdate(cOld)
+	g.Expect(err).To(MatchError(ContainSubstring("parsing comparison version: could not parse \"test\" as version")))
+}
+
```

Comment:
Could we add a couple tests for the positive cases? no upgrade and one minor version bump
I think we are also missing a test for major version upgrade

Reply:
Yes I had checked that the positive case already exists as a test case here
https://github.com/aws/eks-anywhere/blob/main/pkg/api/v1alpha1/cluster_webhook_test.go#L101

Reply:
That's testing minor version bump, what about no version bump?
And we seem to still miss the major version upgrade?

Reply:
These 2 are covered in the `versionskew_test.go` where the main logic lies. Do we want to also test it out again for the cli and controller separately? this will kinda of have duplicate test cases.